src/arch/mod.rs

```diff
@@ -84,7 +89,10 @@ mod tests {
     impl ArchitectureMemory for ArchitectureMemoryDummy {
         fn new<'alloc>(_allocator: &mut impl mm::PageAllocator) -> &'alloc mut Self {
             // We will never use this, we just need the compiler to be happy
-            unsafe { (0 as *mut Self).as_mut().unwrap() }
+            unsafe {
+                #[allow(deref_nullptr)]
+                &mut *core::ptr::null_mut::<Self>()
+            }
```

Comment:
I didn't think about it the first time, but since we're never gonna use it we can use the `unreachable!` builtin macro which coerces properly to any return type
```suggestion
            unreachable!("dummy allocator is not meant to be used");
```
This way we also avoid an extra unsafe block

Reply:
Great idea, fixed